node/src/accountant/mod.rs

```diff
@@ -1385,71 +1383,57 @@ impl PayableThresholdsGauge for PayableThresholdsGaugeReal {
     fn calculate_payout_threshold_in_gwei(
         &self,
         payment_thresholds: &PaymentThresholds,
-        x: u64,
-    ) -> u64 {
-        ThresholdUtils::calculate_sloped_threshold_by_time(payment_thresholds, x) as u64
+        debt_age: u64,
+    ) -> u128 {
+        ThresholdUtils::calculate_finite_debt_limit_by_age(payment_thresholds, debt_age)
     }
     as_any_impl!();
 }
 
 pub struct ThresholdUtils {}
 
 impl ThresholdUtils {
-    pub fn slope(payment_thresholds: &PaymentThresholds, for_wei_computation: bool) -> f64 {
-        let slope = (payment_thresholds.permanent_debt_allowed_gwei as f64
-            - payment_thresholds.debt_threshold_gwei as f64)
-            / Self::convert(payment_thresholds.threshold_interval_sec) as f64;
-        if for_wei_computation {
-            slope * WEIS_OF_GWEI as f64
-        } else {
-            slope
-        }
+    pub fn slope(payment_thresholds: &PaymentThresholds) -> i128 {
+        //remember that any user supplied params must satisfy:
```

Comment:
"Slope is an integer, rather than a float, to improve performance. Since there are computations that divide by the slope, it cannot be allowed to be zero, but since it's an integer, it can't get any closer to zero than -1.

"If the numerator of this computation is less than the denominator, the slope will be calculated as 0; therefore, .permanent_debt_allowed_gwei must be less than .debt_threshold_gwei, so that the numerator will be no greater than -10^9 (-gwei_to_wei(1)), and the denominator must be less than or equal to 10^9.

"If payment_thresholds are ever configurable by the user, these validations should be done on the values before they are accepted."